rskj-core/src/main/java/co/rsk/net/messages/MessageVisitor.java

```diff
@@ -0,0 +1,266 @@
+        }
+
+        if (blockProcessor.hasBlockInSomeBlockchain(block.getHash().getBytes())){
+            logger.trace("Block ignored: it's included in blockchain {} {}", blockNumber, block.getShortHash());
+            return;
+        }
+
+        BlockProcessResult result = this.blockProcessor.processBlock(sender, block);
+        tryRelayBlock(block, result);
+        recordEvent(sender, EventType.VALID_BLOCK);
+    }
+
+    public void apply(StatusMessage message) {
+        final Status status = message.getStatus();
+        logger.trace("Process status {}", status.getBestBlockNumber());
+        this.syncProcessor.processStatus(sender, status);
+    }
+
+    public void apply(GetBlockMessage message) {
+        final byte[] hash = message.getBlockHash();
+        this.blockProcessor.processGetBlock(sender, hash);
+    }
+
+    public void apply(BlockRequestMessage message) {
+        final byte[] hash = message.getBlockHash();
+        this.blockProcessor.processBlockRequest(sender, message.getId(), hash);
+    }
+
+    public void apply(BlockResponseMessage message) {
+        this.syncProcessor.processBlockResponse(sender, message);
+    }
+
+    public void apply(SkeletonRequestMessage message) {
+        final long startNumber = message.getStartNumber();
+        this.blockProcessor.processSkeletonRequest(sender, message.getId(), startNumber);
+    }
+
+    public void apply(BlockHeadersRequestMessage message) {
+        final byte[] hash = message.getHash();
+        final int count = message.getCount();
+        this.blockProcessor.processBlockHeadersRequest(sender, message.getId(), hash, count);
+    }
+
+    public void apply(BlockHashRequestMessage message) {
+        this.blockProcessor.processBlockHashRequest(sender, message.getId(), message.getHeight());
+    }
+
+    public void apply(BlockHashResponseMessage message) {
+        this.syncProcessor.processBlockHashResponse(sender, message);
+    }
+
+    public void apply(NewBlockHashMessage message) {
+        if (blockProcessor.hasBetterBlockToSync()) {
+            loggerMessageProcess.debug("Message[{}] not processed.", message.getMessageType());
+            return;
+        }
+        this.syncProcessor.processNewBlockHash(sender, message);
+    }
+
+    public void apply(SkeletonResponseMessage message) {
+        this.syncProcessor.processSkeletonResponse(sender, message);
+    }
+
+    public void apply(BlockHeadersResponseMessage message) {
+        this.syncProcessor.processBlockHeadersResponse(sender, message);
+    }
+
+    public void apply(BodyRequestMessage message) {
+        final byte[] hash = message.getBlockHash();
+        this.blockProcessor.processBodyRequest(sender, message.getId(), hash);
+    }
+
+    public void apply(BodyResponseMessage message) {
+        this.syncProcessor.processBodyResponse(sender, message);
+    }
+
+    public void apply(NewBlockHashesMessage message) {
+        blockProcessor.processNewBlockHashesMessage(sender, message);
```

Comment:
Missed the `blockProcessor.hasBetterBlockToSync` condition in this message

Reply:
Done!